kotlinx-coroutines-core/common/src/flow/Flow.kt

```diff
@@ -173,6 +173,8 @@ import kotlin.coroutines.*
  * These implementations ensure that the context preservation property is not violated, and prevent most
  * of the developer mistakes related to concurrency, inconsistent flow dispatchers, and cancellation.
  */
+@OptIn(ExperimentalSubclassOptIn::class)
+@SubclassOptInRequired(BrittleForInheritanceCoroutinesApi::class)
```

Comment:
Frankly, taking into account that `Flow` is **the** way to program and that it has been here for a long time, I doubt we ever can change that, and this warning might be just a headache to opt-in. I suggest we might give up on that

Reply:
People who want custom `Flow` implementation should use `AbstractFlow` instead, so I think it's okay to discourage inheriting from `Flow`.

Reply:
I still argue that `Flow` is neither brittle nor experimental.

In my mental model, it's either `Delicate` (because of the invariants that must be preserved) or nothing (because it's quite hard to rationalize it otherwise). WDYT?

Reply:
In my mental model, it's certainly brittle, because it's a mistake to try to implement it without using `AbstractFlow`. It's not `Delicate` because we don't give realistic means of fulfilling the contracts in custom `Flow` instances that don't inherit from `AbstractFlow`.

Reply:
`by`-delegation to `Flow` does happen occasionally (https://grep.app/search?q=%20%3A%20Flow%3C%5B%5E%3E%5D%2B%3E%20by&regexp=true), and the following code does emit a warning:
```kotlin
public class MyFlow: Flow<Int> by flowOf(1, 2, 3) {
}
```

So, `Flow` probably shouldn't be marked as `InternalForInheritance`, at least, as we may want to make it difficult to opt in to that eventually, but this benign pattern should work without issues.